src/main/js/UserMailConfigurationForm.tsx

```diff
@@ -21,65 +21,215 @@
+  };
+
+  const divideTopicsIntoColumns = () => {
+    // We will create two 'column' objects containing the categories with their
+    // corresponding topics.
+    // To get two columns with a more or less equal number of topics, first we
+    // fill up the second column unless we have not put more than the half of
+    // all topics into it. The rest of the topics will go into the first column.
+    // Doing this we will assure that we utilize the columns best and that the
+    // second column is not longer than the first.
+    // To take the category headings into account, we count them as topics.
+    const firstColumn: TopicInCategories = {};
+    const secondColumn: TopicInCategories = {};
+    const groupedTopics = groupTopics();
+    const topicCount = Object.entries(groupedTopics)
+      .map(categoryWithTopics => categoryWithTopics[1])
+      .map(topics => topics.length)
+      .reduce((accumulator, n) => n + accumulator);
+    const lineCount = topicCount + Object.entries(groupedTopics).length;
+
+    let currentCount = 0;
+    Object.entries(groupedTopics).forEach(categoryWithTopics => {
+      if (currentCount + categoryWithTopics[1].length <= lineCount / 2) {
+        currentCount += categoryWithTopics[1].length + 1;
+        secondColumn[categoryWithTopics[0]] = categoryWithTopics[1];
+      }
+    });
+    Object.entries(groupedTopics).forEach(categoryWithTopics => {
+      if (!secondColumn[categoryWithTopics[0]]) {
+        firstColumn[categoryWithTopics[0]] = categoryWithTopics[1];
+      }
+    });
+
+    return [firstColumn, secondColumn];
+  };
+
+  let topics;
+
+  const createTopicColumn = (topicsInCategories: TopicInCategories) => {
     return (
+      <div className={"column is-half"}>
+        {Object.entries(topicsInCategories).map(categoryWithTopics => (
+          <>
+            <label className="label">
+              <Checkbox
+                label={t("mailTopics." + categoryWithTopics[0] + ".label")}
+                name={categoryWithTopics[0]}
+                checked={!!categoryWithTopics[1].find(isTopicSelected)}
+                indeterminate={
+                  categoryWithTopics[1].find(isTopicSelected) &&
+                  categoryWithTopics[1].find(topic => !isTopicSelected(topic))
+                }
+                onChange={categoryChangedHandler(categoryWithTopics[0])}
+              />
+            </label>
+            {categoryWithTopics[1].map(topic => (
+              <Checkbox
+                name={topic.category.name + "/" + topic.name}
+                label={t("mailTopics." + categoryWithTopics[0] + "." + topic.name + ".label")}
+                checked={isTopicSelected(topic)}
+                onChange={topicChangedHandler(topic)}
+                helpText={t("mailTopics." + categoryWithTopics[0] + "." + topic.name + ".helpText")}
+              />
+            ))}
+          </>
+        ))}
+      </div>
+    );
+  };
+
+  if (topicsLoading) {
```

Comment:
I would have initialized topics with null. Also check if loading then return < Loading /> separately from the topics. This looks muddled up.

Reply:
Didn't want to block the whole config view, because the available topics are a part of their own. But it's nothing of importance for me.

Reply:
Okay, that makes sense. I didn't think that through.